core/src/main/java/cucumber/api/event/TimeStampedEvent.java

```diff
@@ -1,14 +1,36 @@
 package cucumber.api.event;
 
 abstract class TimeStampedEvent implements Event {
+
     private final Long timeStamp;
+    private final long timeStampMillis;
 
-    TimeStampedEvent(Long timeStamp) {
+    TimeStampedEvent(Long timeStamp, Long timeStampMillis) {
         this.timeStamp = timeStamp;
+        this.timeStampMillis = timeStampMillis;
     }
 
+    /**
+     * Returns timestamp in nano seconds since an arbitrary start time.
```

Comment:
Any way to make this clearer? I am now wondering what "arbitrary start time" means...

Reply:
Should the JavaDoc here be consistent with the JavaDoc above? "Returns a timestamp in nano seconds."

Reply:
(Or should it be removed if we are using millis everywhere?)

Reply:
I don't think I can be much more clear then "arbitrary start time". Arbitrary X means pick an X, we don't care which one. From the linked `System.nanoTime` documentation.

> Returns the current value of the running Java Virtual Machine's high-resolution time source, in nanoseconds. This method can only be used to measure elapsed time and is not related to any other notion of system or wall-clock time. The value returned represents nanoseconds since some fixed but arbitrary <i>origin</i> time (perhaps in the future, so values may be negative).  The same origin is used by all invocations of this method in an instance of a Java virtual machine; other virtual machine instances are likely to use a different origin.

Can't remove it because it's part of the API so existing plugins will expect it to be present.

Reply:
Good point though. I should have used `{@inheritDoc}`.